mlflow/resource_webhook.go

```diff
@@ -0,0 +1,148 @@
+	var r webhookApiResponse
+	err := a.client.Post(a.context, "/mlflow/registry-webhooks/create", &m, &r)
+	return r.Webhook.ID, err
+}
+
+func (a WebhooksAPI) Read(ID string) (Webhook, error) {
+	var m webhookListResponse
+	err := a.client.Get(a.context, "/mlflow/registry-webhooks/list", nil, &m)
+	if err != nil {
+		return Webhook{}, err
+	}
+	for _, wh := range m.Webhooks {
+		if wh.ID == ID {
+			return wh, nil
+		}
+	}
+	return Webhook{}, fmt.Errorf("Webhook with ID %s isn't found", ID)
+}
+
+// Update the webhook entity
+func (a WebhooksAPI) Update(m Webhook) error {
+	m.ModelName = ""
+	return a.client.Patch(a.context, "/mlflow/registry-webhooks/update", m)
+}
+
+// Delete removes the webhook by its ID
+func (a WebhooksAPI) Delete(ID string) error {
+	return a.client.Delete(a.context, "/mlflow/registry-webhooks/delete", map[string]string{
+		"id": ID,
+	})
+}
+
+func ResourceMLFlowWebhook() *schema.Resource {
+	s := common.StructToSchema(
+		Webhook{},
+		func(m map[string]*schema.Schema) map[string]*schema.Schema {
+			m["status"].ValidateFunc = validation.StringInSlice([]string{"ACTIVE", "TEST_MODE", "DISABLED"}, true)
+			// TODO: do we need a validation for Events?
+			delete(m, "id")
+			p, err := common.SchemaPath(m, "http_url_spec", "enable_ssl_verification")
+			if err == nil {
+				p.Required = false
+				p.Optional = true
+			}
+			p, err = common.SchemaPath(m, "http_url_spec", "url")
+			if err == nil {
+				p.ValidateFunc = validation.IsURLWithHTTPS
+			}
+			m["http_url_spec"].ConflictsWith = []string{"job_spec"}
+			m["job_spec"].ConflictsWith = []string{"http_url_spec"}
+
+			return m
+		})
+
+	return common.Resource{
+		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			var m Webhook
+			common.DataToStructPointer(d, s, &m)
+			if m.HttpUrlSpec == nil && m.JobSpec == nil {
+				return fmt.Errorf("at least one of http_url_spec or job_spec should be specified")
+			}
+
+			hook_id, err := NewWebhooksAPI(ctx, c).Create(m)
+			if err != nil {
+				return err
+			}
+			d.SetId(hook_id)
+			return nil
+		},
+		Read: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			var m_orig Webhook
+			common.DataToStructPointer(d, s, &m_orig)
+			m, err := NewWebhooksAPI(ctx, c).Read(d.Id())
+			if err != nil {
+				return err
+			}
+			// We need to preserve original values as API doesn't return sensitive values
+			if m_orig.JobSpec != nil && m.JobSpec != nil {
+				m.JobSpec.AccessToken = m_orig.JobSpec.AccessToken
+			}
+			if m_orig.HttpUrlSpec != nil && m.HttpUrlSpec != nil {
+				m.HttpUrlSpec.Authorization = m_orig.HttpUrlSpec.Authorization
+			}
+			return common.StructToData(m, s, d)
+		},
+		Update: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			var m Webhook
+			common.DataToStructPointer(d, s, &m)
+			m.ID = d.Id()
+			return NewWebhooksAPI(ctx, c).Update(m)
```

Comment:
For consistency reasons, can you put ID as the first argument to update method?..